src/Reflection/Adapter/ReflectionClass.php

```diff
@@ -366,10 +367,15 @@ public function newInstanceWithoutConstructor()
 
     /**
      * {@inheritDoc}
+     * @throws \LogicException
      */
-    public function newInstanceArgs(?array $args = null)
+    public function newInstanceArgs(array $args = [], ClassLoader $classLoader = null)
```

Comment:
This... is ugly. Especially since we want to maintain reflection compatibility.

I don't think this is the right way to go with this implementation.

Reply:
It's not nice indeed; other option is to provide a default setup for `ClassLoader`, so like:

```php
public function newInstanceArgs(?array $args = null)
{
    $classLoader = new ClassLoader(new EvalLoader(new PhpPrettyPrinter()));
}
```

but I didn't want to get tied to that specific implementation... not sure how else to handle this :s

Reply:
@asgrim add the `ClassLoader` to the `BetterReflection` core or such?

Reply:
That's kinda the problem, using the `EvalLoader` method is crap, the `FileCacheLoader` is much more preferable so should be encouraged, however that needs a folder path specified, so can't be default :/

Reply:
@asgrim I think we should leave the signature be. It is better to not have this thing at all than to have a Frankenstein.